GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -896,6 +912,21 @@ protected override bool ExecuteCommand(int cmd)
             }
         }
 
+        public override IScriptOptionsProvider? GetScriptOptionsProvider(GitModuleControl source)
+        {
+            FileStatusList? fileStatusList = _currentFilesList;
```

Comment:
```suggestion
            Validates.NotNull(_currentFilesList);
            FileStatusList fileStatusList = _currentFilesList;
```

Reply:
This is not guaranteed. The result of this would be randomly throwing an exception on what should be a valid action (Using a hotkey before selecting one of the lists).

The only other place where this validation occurs is in `StartFileHistoryDialog`, and that's because immediately before it checks `if (!Unstaged.Focused && !Staged.Focused)`. If neither is focused, then it is perfectly reasonable for `_currentFilesList` to be `null`.

Reply:
Rather turn `_currentFilesList` into a non-nullable using `Unstaged` as initial value.

Reply:
#11363!

Reply:
Rebased and accepted